### ISSUE TITLE
tzlocal 3.0b1 breaks apscheduler

### ISSUE THREAD

**Issue by apahim** [agronholm/apscheduler#461]
`unleash-client-python` depends on `apscheduler==3.6.3`:
https://github.com/Unleash/unleash-client-python/blob/master/setup.py#L24

`apscheduler` depends on  `tzlocal >= 1.2`:
https://github.com/agronholm/apscheduler/blob/master/setup.cfg#L27

Installing unleash-client-python gets us `tzlocal` 3.0b1 through `apscheduler`. `tzlocal` 3.0b1 breaks the compatibility:

```
  File "/usr/local/lib/python3.6/site-packages/reconcile-0.2.2-py3.6.egg/utils/unleash.py", line 38, in get_feature_toggle_state client.initialize_client()
  File "/usr/local/lib/python3.6/site-packages/UnleashClient-3.4.2-py3.6.egg/UnleashClient/__init__.py", line 141, in initialize_client kwargs=fl_args)
  File "/usr/local/lib/python3.6/site-packages/APScheduler-3.6.3-py3.6.egg/apscheduler/schedulers/base.py", line 443, in add_job self._real_add_job(job, jobstore, replace_existing)
  File "/usr/local/lib/python3.6/site-packages/APScheduler-3.6.3-py3.6.egg/apscheduler/schedulers/base.py", line 859, in _real_add_job replacements['next_run_time'] = job.trigger.get_next_fire_time(None, now)
  File "/usr/local/lib/python3.6/site-packages/APScheduler-3.6.3-py3.6.egg/apscheduler/triggers/interval.py", line 66, in get_next_fire_time
return self.timezone.normalize(next_fire_time)
AttributeError: 'backports.zoneinfo.ZoneInfo' object has no attribute 'normalize'
```
Please either pin-point the tzlocal version you depend on or, at least, pin-point the maximum major version to `tzlocal ~= 2.0`.

**Comment by agronholm** [comment]
How is it that you got a pre-release version installed by default? Are you sure your deployment configuration is alright?

That said, I do plan to pin tzlocal to <3.0 in APScheduler 3.x and >= 3.0 in APScheduler 4.0. I am the author of the breaking changes in tzlocal 3.0 btw :)

**Comment by apahim** [comment]
Nothing special really... you can reproduce with:
```
$ cat setup.py 
from setuptools import find_packages
from setuptools import setup

setup(
    name="test",
    version="0.0.1",
    packages=[],
    install_requires=["apscheduler"]
)
```
```
$ python setup.py install
running install
running bdist_egg
running egg_info
creating test.egg-info
writing test.egg-info/PKG-INFO
writing dependency_links to test.egg-info/dependency_links.txt
writing requirements to test.egg-info/requires.txt
...
Searching for apscheduler
Reading https://pypi.org/simple/apscheduler/
Downloading https://files.pythonhosted.org/packages/f3/34/9ef20ed473c4fd2c3df54ef77a27ae3fc7500b16b192add4720cab8b2c09/APScheduler-3.6.3-py2.py3-none-any.whl#sha256=e8b1ecdb4c7cb2818
913f766d5898183c7cb8936680710a4d3a966e02262e526
Best match: APScheduler 3.6.3
Processing APScheduler-3.6.3-py2.py3-none-any.whl
Installing APScheduler-3.6.3-py2.py3-none-any.whl to /home/apahim/venv_test/lib/python3.8/site-packages
writing requirements to /home/apahim/venv_test/lib/python3.8/site-packages/APScheduler-3.6.3-py3.8.egg/EGG-INFO/requires.txt
Adding APScheduler 3.6.3 to easy-install.pth file

Installed /home/apahim/venv_test/lib/python3.8/site-packages/APScheduler-3.6.3-py3.8.egg
Searching for tzlocal>=1.2
Reading https://pypi.org/simple/tzlocal/
Downloading https://files.pythonhosted.org/packages/79/8c/7d9608dd2643a144bd473405d268973ee1c67aac4c72c4d73910dc8a4754/tzlocal-3.0b1.tar.gz#sha256=efd2821b10836b046fac09c4e31e7b97f45
b89e718e82b8bc18a9e424bb2f662
Best match: tzlocal 3.0b1
Processing tzlocal-3.0b1.tar.gz
Writing /tmp/easy_install-2g2t66n8/tzlocal-3.0b1/setup.cfg
Running tzlocal-3.0b1/setup.py -q bdist_egg --dist-dir /tmp/easy_install-2g2t66n8/tzlocal-3.0b1/egg-dist-tmp-gnnin9nv
/usr/lib64/python3.8/distutils/dist.py:274: UserWarning: Unknown distribution option: 'requires_python'
  warnings.warn(msg)
Moving tzlocal-3.0b1-py3.8.egg to /home/apahim/venv_test/lib/python3.8/site-packages
Adding tzlocal 3.0b1 to easy-install.pth file
...
```

**Comment by agronholm** [comment]
You should not install packages that way. Do `pip install .` if you want to install a package from a local check-out of a codebase. Easy_install (which is deprecated) does not have the safeguards that pip has.

**Comment by apahim** [comment]
Of course. Many old projects still use it though. They used to work, now they don't.
Fix is simple, I just thought you could be interested in knowing about that. Should the tzlocal pre-release be released without changes here, then you would be broken for real.
Anyways, thanks for looking into that.

**Comment by regebro** [comment]
I hope not too many apps will break like this, then we might need to have some sort of smooth transition support.

**Comment by agronholm** [comment]
If app developers don't pin their dependencies somehow then it's their fault if their builds break. Assuming that they can update dependencies across major version changes without ever encountering backwards compatibility issues is simply ludicrous.

**Comment by regebro** [comment]
Sure, but providing them with deprecation warnings and paths is nice.

**Comment by agronholm** [comment]
It's a huge amount of extra work, at least in this case.

**Comment by regebro** [comment]
Could be. Hopefully most users of tzlocal won't require the returned timezone to be from pytz.

**Comment by pganssle** [comment]
> Of course. Many old projects still use it though. They used to work, now they don't.

This seems like a good learning opportunity :)

**Comment by apahim** [comment]
Happy to help.

**Comment by agronholm** [comment]
Fixed by b773c79.
